test/coverage/datamodel/UuidBuilderTest.cpp

```diff
@@ -10,29 +10,261 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <gtest/gtest.h>
-#include <up-cpp/datamodel/builder/Uuid.h>
 
+#include "up-cpp/datamodel/builder/Uuid.h"
 namespace {
 
-class TestFixture : public testing::Test {
+using namespace uprotocol::datamodel::builder;
+
+class TestUuidBuilder : public testing::Test {
 protected:
-	// Run once per TEST_F.
-	// Used to set up clean environments per test.
 	void SetUp() override {}
 	void TearDown() override {}
 
-	// Run once per execution of the test application.
-	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidBuilder() = default;
+	~TestUuidBuilder() = default;
 
-	// Run once per execution of the test application.
-	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+// Test getBuilder
+TEST(UuidBuilderTest, GetBuilder) {
+	auto builder = UuidBuilder::getBuilder();
+	auto uuid = builder.build();
+
+	EXPECT_TRUE(uuid.msb());
+	EXPECT_TRUE(uuid.lsb());
+}
+
+// Test GetTestBuilder
+TEST(UuidBuilderTest, GetTestBuilder) {
+	auto builder = UuidBuilder::getTestBuilder();
+	auto uuid = builder.build();
+
+	EXPECT_TRUE(uuid.msb());
+	EXPECT_TRUE(uuid.lsb());
+}
+
+// Test TestBuilder with time source
+TEST(UuidBuilderTest, WithTimeSource) {
+	auto fixed_time = std::chrono::system_clock::from_time_t(1234567890);
+	auto fixed_time_ms =
+	    std::chrono::time_point_cast<std::chrono::milliseconds>(fixed_time);
+	auto builder = UuidBuilder::getTestBuilder().withTimeSource(
+	    [fixed_time]() { return fixed_time; });
+	auto uuid = builder.build();
+
+	EXPECT_EQ(uuid.msb() >> 16, fixed_time_ms.time_since_epoch().count());
+}
+
+// Test RandomSource
+TEST(UuidBuilderTest, WithRandomSource) {
+	auto fixed_random = 0x1234567890ABCDEF;
+	auto builder = UuidBuilder::getTestBuilder().withRandomSource(
+	    [fixed_random]() { return fixed_random; });
+	auto uuid = builder.build();
+
+	EXPECT_EQ(uuid.lsb() & 0x3FFFFFFFFFFFFFFF,
+	          fixed_random & 0x3FFFFFFFFFFFFFFF);
+}
+
+// Test independent state
+TEST(UuidBuilderTest, WithIndependentState) {
```

Comment:
Add a copy of this test for `WithDependentState` where you verify that two regular builders produce values with related states. For example, the `EXPECT_NE` on line 89 would turn into an `EXPECT_EQ`

Reply:
Not worrying about shared state for this PR